### ISSUE TITLE
🚧Add resize capture in scenes

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/Velaptor#954]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

Add the ability to get updates on when the window size has changed in the scene system.

This will allow users to know when the window size has changed in a scene. This will involve adding a new `virtual` method to the `SceneBase` class, which will implement the new method that will be added to the `IScene` interface.

Also, this will need to be added to the `ISceneManager` interface.

1. Add a new method to the `IScene` interface named `Resize(SizeU size)`
    - Code Docs
      ```html
      /// <summary>
      /// Updates the <see cref="WindowSize"/>.
      /// </summary>
      /// <param name="size">The new size.</param>
      ```
3. Add a new `virtual` method implementation to the `SceneBase` class
    - This is what will get invoked by the `internal SceneManager` class
    - This will use `/// <inheritdoc/>` for the code docs
4. Add a new method to the `ISceneManager` interface named `Resize(SizeU size)`
      - Code docs
        ```html
        /// <summary>
        /// Updates the size of the window for each scene.
        /// </summary>
        /// <param name="size">The new size.</param>
        ```
6. Add the new method implementation with the name `Resize(SizeU size)` to the `internal SceneManager` class.
    - This method will be invoked by the `Window` class in the `protected virtual void OnResize(SizeU size)` method
    - This will use `/// <inheritdoc/>` for the code docs
7. Update the scene guide on the main docs (This will be done by @CalvinWilkinson once this issue is complete)

> [!Note]
> Coding Standards:
> 1. Make sure to use [expression bodied methods](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/statements-expressions-operators/expression-bodied-members) when there is only a single statement for methods.

### Acceptance Criteria

```[tasklist]
### The items to complete to satisfy the Definition of Done.
- [x] All unit tests executed and passed locally after PR work is complete. _(Why? Refer to 'Additional Information' below.)_
- [x] New method named `Resize()` added to the `IScene` interface
- [x] Method implementation of `IScene.Resize()` added to the `SceneBase` class
- [x] New method named `Resize()` added to the `ISceneManager` interface
- [x] Method implementation of `ISceneManager.Resize()` added to the `SceneManager` class
- [x] Scene guide updated on the main doc website
```

### ToDo Items

```[tasklist]
### The items to complete to satisfy the Definition of Done.
- [x] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [x] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [x] Issue linked to the correct milestone _(if applicable)_.
```

### Additional Information:

**_<details closed><summary>Unit Tests</summary>_**

Reasons for local unit test execution:
- Unit tests might pass locally but not in the CI environment during the status check process or vice-versa.
- Tests might pass on the developer's machine but not necessarily on the code reviewer's machine.
- If you notice that the test status check has passed but the test failed locally, please notify a project maintainer!

💡Warning💡
If the unit tests pass remotely and are not executed locally, this means we could be letting a bug slip into production.
Though bugs will always exist in some capacity, we should all do our part to help prevent them from happening.
</details>

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type              | Label                     |
|:-------------------------|:--------------------------|
| Bug Fixes                | `🐛bug`                   |
| Breaking Changes         | `🧨breaking changes`      |
| New Feature              | `✨new feature`           |
| CICD Changes             | `♻️cicd`                  |
| Config Changes           | `⚙️config`                |
| Performance Improvements | `🏎️performance`           |
| Code Doc Changes         | `🗒️documentation/code`    |
| Product Doc Changes      | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

**Comment by sbundaz** [comment]
I can take it.
